plugins/network-generator/src/main/kotlin/dev/icerock/moko/network/MultiPlatformNetworkGeneratorPlugin.kt

```diff
@@ -16,28 +16,54 @@ class MultiPlatformNetworkGeneratorPlugin : Plugin<Project> {
     private val openApiGenerator = OpenApiGeneratorPlugin()
 
     override fun apply(target: Project) {
-        openApiGenerator.apply(target)
 
-        val generatedDir = "${target.buildDir}/generate-resources/main"
+        val mokoNetworkExtension = target.extensions.create("mokoNetwork", SpecConfig::class.java)
 
         target.afterEvaluate {
-            extensions.findByType(KotlinMultiplatformExtension::class.java)?.run {
-                val sourceSet = sourceSets.getByName(KotlinSourceSet.COMMON_MAIN_SOURCE_SET_NAME)
-                val sources = "$generatedDir/src/main/kotlin"
-                sourceSet.kotlin.srcDir(sources)
-            }
 
-            val removeGeneratedCodeTask =
-                tasks.create("removeGeneratedOpenApiCode", Delete::class) {
-                    delete(file(generatedDir))
+            mokoNetworkExtension.specs.forEach { spec ->
+
+                val generatedDir =
+                    "${target.buildDir}/generate-resources/main/src/main/kotlin/${spec.name}"
+
+                extensions.findByType(KotlinMultiplatformExtension::class.java)?.run {
+                    val sourceSet =
+                        sourceSets.getByName(KotlinSourceSet.COMMON_MAIN_SOURCE_SET_NAME)
+                    val sources = generatedDir
+                    sourceSet.kotlin.srcDir(sources)
                 }
 
-            tasks.findByName("openApiGenerate")?.let {
-                it.dependsOn(removeGeneratedCodeTask)
+                val generateTask = tasks.create(
+                    "${spec.name}OpenApiGenerate",
+                    org.openapitools.generator.gradle.plugin.tasks.GenerateTask::class.java
+                ) {
+                    group = "openapi"
+
+                    inputSpec.set(spec.inputSpec?.path)
+                    packageName.set(spec.packageName)
+
+                    additionalProperties.set(
+                        mutableMapOf(
+                            "nonPublicApi" to "${spec.isInternal}",
+                            "openApiClasses" to "${spec.isOpen}"
+                        )
+                    )
+
+                    outputDir.set(generatedDir)
+                    generatorName.set("kotlin-ktor-client")
+                    spec.configureTask?.invoke(this)
+                }
+
+                val removeGeneratedCodeTask =
+                    tasks.create("${spec.name}removeGeneratedOpenApiCode", Delete::class) {
```

Comment:
snakeCase - Remove should start from uppercase. now you got `petstoreremoveGeneratedOpenApiCode`